packages/typespec-ts/test/integration/scalar.spec.ts

```diff
@@ -77,4 +78,118 @@ describe("Scalar Client", () => {
       assert.fail(err as string);
     }
   });
+
+  it("should get decimal response body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal/response_body")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual(result.body, 0.33333);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put decimal request body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal/resquest_body")
+        .put({ body: 0.33333 });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get decimal request parameter", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal/request_parameter")
+        .get({ queryParameters: { value: 0.33333 } });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get decimal128 response body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal128/response_body")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual(result.body, 0.33333);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put decimal128 request body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal128/resquest_body")
+        .put({ body: 0.33333 });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get decimal128 request parameter", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal128/request_parameter")
+        .get({ queryParameters: { value: 0.33333 } });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should post decimal verify", async () => {
+    try {
+      // prepare the verification
+      const getResult = await client
+        .path("/type/scalar/decimal/prepare_verify")
+        .get();
+      // do any calculation based on third party library
+      let total = new Decimal(0);
```

Comment:
Include `decimal.js` for calculation.

Reply:
Is this meant to be a test case for round trip ? and because we have precision loss, we have to leverage the third party library ? I thought we already have precison loss, when we get the data from json.parse ?

Reply:
There are two kinds of risks - 1) calculation and 2) serialization/deserialization. You could see detailed [discussion here](https://github.com/Azure/autorest.typescript/issues/2157#issuecomment-1850671951).

Here is mainly for cadl-ranch test case for calculation, in that case we need three decimals to add together and then the sum would be used to verify in server side.

Reply:
Thanks for the detailed explaination, I am not quite sure if it's necessary to import the decimal.js here? Just wondering if there's any case that toFixed could not work but decimal.js can, giving the result and the added numbers are all primitive number type ? 
@joheredi @xirzec any thoughts here ?

Reply:
I may not get your point of toFixed. Let me explain the case more, in cadl-ranch we would get three decimals they are 0.1, 0.1 and 0.1. Then we need to add these three decimals and then send the total `0.3` back to server side.

In JavaScript directly adding them would lose the precision and we got `0.30000000000000004`.

```js
const a = 0.1 +0.1 +0.1;
console.log(a, a.toFixed());
// [LOG]: 0.30000000000000004,  "0" 
```

Including the `decimal.js` as a dev dependency is trying to simulate customers' behavior who really cares about precision for any calculation.

Reply:
In the first example you are missing the digit parameter of toFixed

```ts
const total = 0.1 + 0.1 + 0.1;
console.log(total, total.toFixed(1));
// [LOG]: 0.30000000000000004,  "0.3" 
```

I think we should try not to add the external library unless completly necessary

Reply:
There are two concerns, 1) toFixed would return a string; do we need to castcade to number? 2) the test case is trying to simulate the items preparation from server side during runtime then how do we know which digits we sould use in toFixed?

Reply:
@joheredi is right about my previous point, my another point is that even if we need to use decimal.js to handle the calculation as of the result out of number range, because we can only accept number result, the decimal.js toNumber will also have precision loss. how is that precision loss different than using toFixed ?

Reply:
One suggestion:  Can we just send 0.30000000000000004 to server which is the actual result and let service side to return error code. 
We could still verify the 'failure' error code or exception and add a comment that this case failed because we will lose the precision when doing the calculation and this is a known issue in JS.